test/e2e/addon/podidentityaddon.go

```diff
@@ -0,0 +1,66 @@
+package addon
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/go-logr/logr"
+	clientgo "k8s.io/client-go/kubernetes"
+
+	"github.com/aws/eks-hybrid/test/e2e/errors"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+)
+
+type PodIdentityAddon struct {
+	Addon
+	kubernetes *clientgo.Clientset
+	iamClient  *iam.Client
+	roleArn    string
+}
+
+const (
+	podIdentityServiceAccount = "pod-identity-sa"
+	namespace                 = "default"
+)
+
+func NewPodIdentityAddon(cluster, name string, k8sClient *clientgo.Clientset, iamClient *iam.Client, roleArn string) PodIdentityAddon {
+	return PodIdentityAddon{
+		Addon: Addon{
+			Cluster:       cluster,
+			Name:          name,
+			Configuration: "{\"daemonsets\":{\"hybrid\":{\"create\": true}}}",
+		},
+		kubernetes: k8sClient,
+		iamClient:  iamClient,
+		roleArn:    roleArn,
+	}
+}
+
+func (p PodIdentityAddon) Create(ctx context.Context, client *eks.Client, logger logr.Logger) error {
```

Comment:
why is the eks client passed an argument here but the iam client and k8s client passed to the constructor?

i would vote to pass them all to this method (if you need them) and make the struct "data only"